core/types/block.libevm_test.go

```diff
@@ -66,14 +82,36 @@ func TestHeaderHooks(t *testing.T) {
 	extras := RegisterExtras[stubHeaderHooks, *stubHeaderHooks, struct{}]()
 	rng := ethtest.NewPseudoRand(13579)
 
-	t.Run("EncodeRLP", func(t *testing.T) {
-		suffix := rng.Bytes(8)
+	suffix := rng.Bytes(8)
+	hdr := &Header{
+		ParentHash: rng.Hash(),
+	}
+	extras.Header.Get(hdr).suffix = append([]byte{}, suffix...)
+
+	t.Run("MarshalJSON", func(t *testing.T) {
+		got, err := json.Marshal(hdr)
+		require.NoError(t, err, "json.Marshal(%T)", hdr)
+		assert.Equal(t, fakeHeaderJSON(hdr, suffix), got)
+	})
 
-		hdr := &Header{
-			ParentHash: rng.Hash(),
+	t.Run("UnmarshalJSON", func(t *testing.T) {
+		hdr := new(Header)
+		stub := &stubHeaderHooks{
+			setHeaderToOnUnmarshalOrDecode: Header{
+				Extra: []byte("can you solve this puzzle? 0xbda01b6cf56c303bd3f581599c0d5c0b"),
```

Comment:
🤔 🤔 🤔 🤔 🤔 🤔 🤔 🤔 🤔 🤔 🤔 🤔 🤔 🤔 🤔 🤔 🤔 🤔 🤔 🤔

Reply:
Well?

Reply:
Any hint on an encoding?
Is this your 256 bits private key? 😄